code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/CarouselTemplateUtil.kt

```diff
@@ -0,0 +1,115 @@
+        var newRightIndex = 0
+        Log.trace(
+            PushTemplateConstants.LOG_TAG,
+            SELF_TAG,
+            "Current center index is $centerIndex and list size is $listSize."
+        )
+        if ((action == PushTemplateConstants.IntentActions.FILMSTRIP_LEFT_CLICKED) || (action == PushTemplateConstants.IntentActions.MANUAL_CAROUSEL_LEFT_CLICKED)) {
+            newCenterIndex = (centerIndex - 1 + listSize) % listSize
+            newLeftIndex = (newCenterIndex - 1 + listSize) % listSize
+            newRightIndex = centerIndex
+        } else if ((
+            (action == PushTemplateConstants.IntentActions.FILMSTRIP_RIGHT_CLICKED) || (
+                action ==
+                    PushTemplateConstants.IntentActions.MANUAL_CAROUSEL_RIGHT_CLICKED
+                )
+            )
+        ) {
+            newCenterIndex = (centerIndex + 1) % listSize
+            newLeftIndex = centerIndex
+            newRightIndex = (newCenterIndex + 1) % listSize
+        }
+        newIndices.add(newLeftIndex)
+        newIndices.add(newCenterIndex)
+        newIndices.add(newRightIndex)
+        Log.trace(
+            PushTemplateConstants.LOG_TAG,
+            SELF_TAG,
+            "Calculated new indices. New center index is $newCenterIndex, new left index is $newLeftIndex, and new right index is $newRightIndex."
+        )
+        return newIndices
+    }
+
+    internal fun parseCarouselItems(carouselItemsString: String?): MutableList<CarouselPushTemplate.CarouselItem> {
+        val carouselItems = mutableListOf<CarouselPushTemplate.CarouselItem>()
+        if (carouselItemsString.isNullOrEmpty()) {
+            Log.debug(
+                PushTemplateConstants.LOG_TAG,
+                SELF_TAG,
+                "No carousel items found in the push template."
+            )
+            return carouselItems
+        }
+        try {
+            val jsonArray = JSONArray(carouselItemsString)
+            for (i in 0 until jsonArray.length()) {
+                val item = jsonArray.getJSONObject(i)
+                val imageUri = item.getString(PushTemplateConstants.CarouselItemKeys.IMAGE)
+                val captionText = item.optString(PushTemplateConstants.CarouselItemKeys.TEXT, "")
+                val interactionUri = item.optString(PushTemplateConstants.CarouselItemKeys.URL, "")
+                carouselItems.add(
+                    CarouselPushTemplate.CarouselItem(
+                        imageUri,
+                        captionText,
+                        interactionUri
+                    )
+                )
+            }
+        } catch (e: JSONException) {
+            Log.debug(
+                PushTemplateConstants.LOG_TAG,
+                SELF_TAG,
+                "Failed to parse carousel items from the push template: ${e.localizedMessage}"
+            )
+        }
+        return carouselItems
+    }
+
+    internal fun getDefaultCarouselIndex(carouselLayoutType: String): Int {
```

Comment:
Since this is only used for `ManualCarouselTemplate` can you make it a utility in its companion object.